lib/foreman_tasks/tasks/export_tasks.rake

```diff
@@ -242,15 +242,16 @@ namespace :foreman_tasks do
     export_filename = ENV['TASK_FILE'] || "/tmp/task-export-#{Time.now.to_i}.#{format == 'csv' ? 'csv' : 'tar.gz'}"
 
     tasks = ForemanTasks::Task.search_for(filter)
+    tasks = tasks.merge(ForemanTasks::Task.with_duration.distinct)
```

Comment:
Here again `tasks = ForemanTasks::Task.search_for(filter).with_duration.distinct` should work